### ISSUE TITLE
Generate the second picture, and the generation history will immediately eliminate the first picture.

### ISSUE THREAD

**Issue by 1025307336** [mcmonkeyprojects/SwarmUI#157]
### Expected Behavior

The second picture should have been generated, but the first picture will also be retained in the generation history, and the main picture preview will jump to the newly generated picture.

### Actual Behavior

![QQ20240812-124241](https://github.com/user-attachments/assets/08603d6a-ae14-407a-8915-84c1a8ebf087)

At first, I redeployed a swarm on another disk and transferred the models, images and comfy to the new swarm.

But the problem that occurred the first time was that the new image after generation would not automatically switch to the new image. Then I cleared all the browser cache and other data. The problem that occurred the second time became that after clicking Generate, the previous image The history of the picture disappears directly

Later, no matter how I redeployed it to other disks, this problem still occurred.

I personally feel that the three automatic function switches in the upper right corner have lost their function and cannot complete the settings.

### Steps to Reproduce

The problem occurred with my steps above, but I don't know if it can be reproduced on other computers.

### Debug Logs

```powershell
2024-08-12 12:36:45.704 [Info] Generated an image in 1.63 (prep) and 5.38 (gen) seconds
2024-08-12 12:37:19.515 [Debug] [ComfyUI-0/STDOUT] [ComfyUI-Manager] default cache updated: https://raw.githubusercontent.com/ltdrdata/ComfyUI-Manager/main/alter-list.json
2024-08-12 12:38:54.623 [Debug] [ComfyUI-0/STDERR] Exception in thread Thread-6 (<lambda>):
2024-08-12 12:38:54.624 [Warning] [ComfyUI-0/STDERR] Traceback (most recent call last):
2024-08-12 12:38:54.625 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\aiohttp\client_proto.py", line 94, in connection_lost
2024-08-12 12:38:54.634 [Warning] [ComfyUI-0/STDERR]     uncompleted = self._parser.feed_eof()
2024-08-12 12:38:54.635 [Warning] [ComfyUI-0/STDERR]                   ^^^^^^^^^^^^^^^^^^^^^^^
2024-08-12 12:38:54.636 [Warning] [ComfyUI-0/STDERR]   File "aiohttp\\_http_parser.pyx", line 510, in aiohttp._http_parser.HttpParser.feed_eof
2024-08-12 12:38:54.639 [Warning] [ComfyUI-0/STDERR] aiohttp.http_exceptions.ContentLengthError: 400, message:
2024-08-12 12:38:54.640 [Warning] [ComfyUI-0/STDERR]   Not enough data for satisfy content length header.
2024-08-12 12:38:54.641 [Warning] [ComfyUI-0/STDERR] 
2024-08-12 12:38:54.642 [Debug] [ComfyUI-0/STDERR] The above exception was the direct cause of the following exception:
2024-08-12 12:38:54.642 [Debug] [ComfyUI-0/STDERR] 
2024-08-12 12:38:54.642 [Warning] [ComfyUI-0/STDERR] Traceback (most recent call last):
2024-08-12 12:38:54.643 [Warning] [ComfyUI-0/STDERR]   File "threading.py", line 1045, in _bootstrap_inner
2024-08-12 12:38:54.643 [Warning] [ComfyUI-0/STDERR]   File "threading.py", line 982, in run
2024-08-12 12:38:54.647 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-Manager\glob\manager_server.py", line 1694, in <lambda>
2024-08-12 12:38:54.648 [Warning] [ComfyUI-0/STDERR]     threading.Thread(target=lambda: asyncio.run(default_cache_update())).start()
2024-08-12 12:38:54.650 [Warning] [ComfyUI-0/STDERR]                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-08-12 12:38:54.650 [Warning] [ComfyUI-0/STDERR]   File "asyncio\runners.py", line 190, in run
2024-08-12 12:38:54.653 [Warning] [ComfyUI-0/STDERR]   File "asyncio\runners.py", line 118, in run
2024-08-12 12:38:54.656 [Warning] [ComfyUI-0/STDERR]   File "asyncio\base_events.py", line 654, in run_until_complete
2024-08-12 12:38:54.658 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-Manager\glob\manager_server.py", line 1691, in default_cache_update
2024-08-12 12:38:54.659 [Warning] [ComfyUI-0/STDERR]     await asyncio.gather(a, b, c, d, e)
2024-08-12 12:38:54.660 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-Manager\glob\manager_server.py", line 1678, in get_cache
2024-08-12 12:38:54.660 [Warning] [ComfyUI-0/STDERR]     json_obj = await core.get_data(uri, True)
2024-08-12 12:38:54.661 [Warning] [ComfyUI-0/STDERR]                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-08-12 12:38:54.662 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\ComfyUI\custom_nodes\ComfyUI-Manager\glob\manager_core.py", line 584, in get_data
2024-08-12 12:38:54.662 [Warning] [ComfyUI-0/STDERR]     json_text = await resp.text()
2024-08-12 12:38:54.663 [Warning] [ComfyUI-0/STDERR]                 ^^^^^^^^^^^^^^^^^
2024-08-12 12:38:54.664 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\aiohttp\client_reqrep.py", line 1153, in text
2024-08-12 12:38:54.664 [Warning] [ComfyUI-0/STDERR]     await self.read()
2024-08-12 12:38:54.665 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\aiohttp\client_reqrep.py", line 1111, in read
2024-08-12 12:38:54.665 [Warning] [ComfyUI-0/STDERR]     self._body = await self.content.read()
2024-08-12 12:38:54.666 [Warning] [ComfyUI-0/STDERR]                  ^^^^^^^^^^^^^^^^^^^^^^^^^
2024-08-12 12:38:54.667 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\aiohttp\streams.py", line 383, in read
2024-08-12 12:38:54.667 [Warning] [ComfyUI-0/STDERR]     block = await self.readany()
2024-08-12 12:38:54.668 [Warning] [ComfyUI-0/STDERR]             ^^^^^^^^^^^^^^^^^^^^
2024-08-12 12:38:54.669 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\aiohttp\streams.py", line 405, in readany
2024-08-12 12:38:54.669 [Warning] [ComfyUI-0/STDERR]     await self._wait("readany")
2024-08-12 12:38:54.670 [Warning] [ComfyUI-0/STDERR]   File "E:\SwarmUI\dlbackend\comfy\python_embeded\Lib\site-packages\aiohttp\streams.py", line 312, in _wait
2024-08-12 12:38:54.670 [Warning] [ComfyUI-0/STDERR]     await waiter
2024-08-12 12:38:54.671 [Warning] [ComfyUI-0/STDERR] aiohttp.client_exceptions.ClientPayloadError: Response payload is not completed: <ContentLengthError: 400, message='Not enough data for satisfy content length header.'>




2024-08-12 12:51:23.309 [Debug] [ComfyUI-0/STDERR] Prompt executed in 5.06 seconds
2024-08-12 12:51:23.473 [Debug] [ComfyUI-0/STDERR] 
2024-08-12 12:51:23.604 [Debug] [ComfyUI-0/STDERR]   0%|          | 0/20 [00:00<?, ?it/s]
2024-08-12 12:51:23.737 [Debug] [ComfyUI-0/STDERR]  10%|鈻         | 2/20 [00:00<00:01, 15.26it/s]
2024-08-12 12:51:23.883 [Debug] [ComfyUI-0/STDERR]  20%|鈻堚枅        | 4/20 [00:00<00:01, 15.17it/s]
2024-08-12 12:51:24.058 [Debug] [ComfyUI-0/STDERR]  30%|鈻堚枅鈻       | 6/20 [00:00<00:00, 14.47it/s]
2024-08-12 12:51:24.235 [Debug] [ComfyUI-0/STDERR]  40%|鈻堚枅鈻堚枅      | 8/20 [00:00<00:00, 13.07it/s]
2024-08-12 12:51:24.264 [Info] Generated an image in 7.89 (prep) and 5.20 (gen) seconds
2024-08-12 12:51:24.416 [Debug] [ComfyUI-0/STDERR]  50%|鈻堚枅鈻堚枅鈻     | 10/20 [00:00<00:00, 12.40it/s]
2024-08-12 12:51:24.605 [Debug] [ComfyUI-0/STDERR]  60%|鈻堚枅鈻堚枅鈻堚枅    | 12/20 [00:00<00:00, 11.89it/s]
2024-08-12 12:51:24.740 [Debug] [ComfyUI-0/STDERR]  70%|鈻堚枅鈻堚枅鈻堚枅鈻   | 14/20 [00:01<00:00, 11.44it/s]
2024-08-12 12:51:24.879 [Debug] [ComfyUI-0/STDERR]  80%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅  | 16/20 [00:01<00:00, 12.32it/s]
2024-08-12 12:51:25.013 [Debug] [ComfyUI-0/STDERR]  90%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻 | 18/20 [00:01<00:00, 12.90it/s]
2024-08-12 12:51:25.013 [Debug] [ComfyUI-0/STDERR] 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 20/20 [00:01<00:00, 13.46it/s]
2024-08-12 12:51:25.013 [Debug] [ComfyUI-0/STDERR] 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 20/20 [00:01<00:00, 12.99it/s]
2024-08-12 12:51:25.455 [Debug] [ComfyUI-0/STDERR] 
2024-08-12 12:51:25.818 [Debug] [ComfyUI-0/STDERR]   0%|          | 0/10 [00:00<?, ?it/s]
2024-08-12 12:51:26.128 [Debug] [ComfyUI-0/STDERR]  10%|鈻         | 1/10 [00:00<00:03,  2.76it/s]
2024-08-12 12:51:26.427 [Debug] [ComfyUI-0/STDERR]  20%|鈻堚枅        | 2/10 [00:00<00:02,  3.02it/s]
2024-08-12 12:51:26.800 [Debug] [ComfyUI-0/STDERR]  30%|鈻堚枅鈻       | 3/10 [00:00<00:02,  3.16it/s]
2024-08-12 12:51:27.156 [Debug] [ComfyUI-0/STDERR]  40%|鈻堚枅鈻堚枅      | 4/10 [00:01<00:02,  2.95it/s]
2024-08-12 12:51:27.507 [Debug] [ComfyUI-0/STDERR]  50%|鈻堚枅鈻堚枅鈻     | 5/10 [00:01<00:01,  2.90it/s]
2024-08-12 12:51:27.793 [Debug] [ComfyUI-0/STDERR]  60%|鈻堚枅鈻堚枅鈻堚枅    | 6/10 [00:02<00:01,  2.88it/s]
2024-08-12 12:51:28.080 [Debug] [ComfyUI-0/STDERR]  70%|鈻堚枅鈻堚枅鈻堚枅鈻   | 7/10 [00:02<00:00,  3.06it/s]
2024-08-12 12:51:28.362 [Debug] [ComfyUI-0/STDERR]  80%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅  | 8/10 [00:02<00:00,  3.18it/s]
2024-08-12 12:51:28.634 [Debug] [ComfyUI-0/STDERR]  90%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻 | 9/10 [00:02<00:00,  3.29it/s]
2024-08-12 12:51:28.634 [Debug] [ComfyUI-0/STDERR] 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 10/10 [00:03<00:00,  3.40it/s]
2024-08-12 12:51:28.634 [Debug] [ComfyUI-0/STDERR] 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 10/10 [00:03<00:00,  3.15it/s]
2024-08-12 12:51:28.988 [Debug] [ComfyUI-0/STDERR] Prompt executed in 5.53 seconds
2024-08-12 12:51:29.739 [Info] Generated an image in 8.88 (prep) and 5.53 (gen) seconds
2024-08-12 12:53:41.634 [Debug] [ComfyUI-0/STDOUT] Error. No naistyles.csv found. Put your naistyles.csv in the custom_nodes/ComfyUI_NAI-mod/CSV directory of ComfyUI. Then press "Refresh".
2024-08-12 12:53:41.634 [Debug] [ComfyUI-0/STDOUT]                   Your current root directory is: E:\SwarmUI\dlbackend\comfy\ComfyUI
2024-08-12 12:53:41.634 [Debug] [ComfyUI-0/STDOUT]             
2024-08-12 12:53:41.635 [Debug] [ComfyUI-0/STDOUT] Error. No naistyles.csv found. Put your naistyles.csv in the custom_nodes/ComfyUI_NAI-mod/CSV directory of ComfyUI. Then press "Refresh".
2024-08-12 12:53:41.635 [Debug] [ComfyUI-0/STDOUT]                   Your current root directory is: E:\SwarmUI\dlbackend\comfy\ComfyUI
2024-08-12 12:53:41.635 [Debug] [ComfyUI-0/STDOUT]             
2024-08-12 12:53:41.635 [Debug] [ComfyUI-0/STDOUT] Error. No naistyles.csv found. Put your naistyles.csv in the custom_nodes/ComfyUI_NAI-mod/CSV directory of ComfyUI. Then press "Refresh".
2024-08-12 12:53:41.635 [Debug] [ComfyUI-0/STDOUT]                   Your current root directory is: E:\SwarmUI\dlbackend\comfy\ComfyUI
2024-08-12 12:53:41.635 [Debug] [ComfyUI-0/STDOUT]             
2024-08-12 12:53:41.831 [Debug] [Load ComfyUI Self-Starting #0] Got valid value set, will parse...
2024-08-12 12:53:41.831 [Debug] Comfy backend 0 using model folder format: backslash \ due to model 3D模型\CandyApple.safetensors
2024-08-12 12:53:41.831 [Debug] [Load ComfyUI Self-Starting #0] Done parsing value set.
2024-08-12 12:53:42.496 [Debug] Data refreshed!
2024-08-12 12:53:48.808 [Info] User local requested 1 image with model '3D模型/pvcSTYLE_v10.safetensors'...
2024-08-12 12:53:48.810 [Debug] [BackendHandler] Backend request #13 for model 3D模型/pvcSTYLE_v10.safetensors, maxWait=7.00:00:00.
2024-08-12 12:53:48.810 [Debug] [BackendHandler] Backend request #13 found correct model on #0
2024-08-12 12:53:48.810 [Debug] [BackendHandler] Backend request #13 finished.
2024-08-12 12:53:48.814 [Debug] [ComfyUI-0/STDERR] got prompt
2024-08-12 12:53:48.836 [Debug] [ComfyUI-0/STDERR] 
2024-08-12 12:53:49.116 [Debug] [ComfyUI-0/STDERR]   0%|          | 0/20 [00:00<?, ?it/s]
2024-08-12 12:53:49.269 [Debug] [ComfyUI-0/STDERR]   5%|鈻         | 1/20 [00:00<00:05,  3.57it/s]
2024-08-12 12:53:49.409 [Debug] [ComfyUI-0/STDERR]  15%|鈻堚枌        | 3/20 [00:00<00:02,  7.74it/s]
2024-08-12 12:53:49.545 [Debug] [ComfyUI-0/STDERR]  25%|鈻堚枅鈻       | 5/20 [00:00<00:01, 10.12it/s]
2024-08-12 12:53:49.680 [Debug] [ComfyUI-0/STDERR]  35%|鈻堚枅鈻堚枌      | 7/20 [00:00<00:01, 11.64it/s]
2024-08-12 12:53:49.824 [Debug] [ComfyUI-0/STDERR]  45%|鈻堚枅鈻堚枅鈻     | 9/20 [00:00<00:00, 12.67it/s]
2024-08-12 12:53:49.972 [Debug] [ComfyUI-0/STDERR]  55%|鈻堚枅鈻堚枅鈻堚枌    | 11/20 [00:00<00:00, 13.10it/s]
2024-08-12 12:53:50.112 [Debug] [ComfyUI-0/STDERR]  65%|鈻堚枅鈻堚枅鈻堚枅鈻   | 13/20 [00:01<00:00, 13.21it/s]
2024-08-12 12:53:50.240 [Debug] [ComfyUI-0/STDERR]  75%|鈻堚枅鈻堚枅鈻堚枅鈻堚枌  | 15/20 [00:01<00:00, 13.53it/s]
2024-08-12 12:53:50.363 [Debug] [ComfyUI-0/STDERR]  85%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻 | 17/20 [00:01<00:00, 14.15it/s]
2024-08-12 12:53:50.421 [Debug] [ComfyUI-0/STDERR]  95%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枌| 19/20 [00:01<00:00, 14.75it/s]
2024-08-12 12:53:50.421 [Debug] [ComfyUI-0/STDERR] 100%|鈻堚枅鈻堚枅鈻堚枅鈻堚枅鈻堚枅| 20/20 [00:01<00:00, 12.61it/s]
2024-08-12 12:53:50.714 [Debug] [ComfyUI-0/STDERR] 
2024-08-12 12:53:50.982 [Debug] [ComfyUI-0/STDERR]   0%|          | 0/10 [00:00<?, ?it/s]
2024-08-12 12:53:51.263 [Debug] [ComfyUI-0/STDERR]  10%|鈻         | 1/10 [00:00<00:02,  3.72it/s]
2024-08-12 12:53:51.547 [Debug] [ComfyUI-0/STDERR]  20%|鈻堚枅        | 2/10 [00:00<00:02,  3.63it/s]
2024-08-12 12:53:51.831 [Debug] [ComfyUI-0/STDERR]  30%|鈻堚枅鈻       | 3/10 [00:00<00:01,  3.58it/s]
2024-08-12 12:53:52.116 [Debug] [ComfyUI-0/STDERR]  40%|鈻堚枅鈻堚枅      | 4/10 [00:01<00:01,  3.56it/s]
2024-08-12 12:53:52.289 [Info] User local requested 1 image with model '3D模型/pvcSTYLE_v10.safetensors'...
2024-08-12 12:53:52.290 [Debug] [BackendHandler] Backend request #14 for model 3D模型/pvcSTYLE_v10.safetensors, maxWait=7.00:00:00.
2024-08-12 12:53:52.290 [Debug] [BackendHandler] Backend request #14 found correct model on #0
2024-08-12 12:53:52.290 [Debug] [BackendHandler] Backend request #14 finished.
2024-08-12 12:53:52.291 [Debug] [ComfyUI-0/STDERR]  50%|鈻堚枅鈻堚枅鈻     | 5/10 [00:01<00:01,  3.54it/s]got prompt
2024-08-12 12:53:52.408 [Debug] [ComfyUI-0/STDERR] 
2024-08-12 12:53:52.431 [Info] User local requested 1 image with model '3D模型/pvcSTYLE_v10.safetensors'...
```

### Other

I tried various browsers but it didn't work, but I don't understand why multiple swarms from different disks share the same UI interface settings after startup if they are started separately.

**Comment by mcmonkey4eva** [comment]
`I don't understand why multiple swarms from different disks share the same UI interface settings after startup if they are started separately.`
If you're launching them all on the same machine with same port, they share some settings because some interface controls are in browser cache. You can give your different installs different ports to dodge this if you want them separate.

`But the problem that occurred the first time was that the new image after generation would not automatically switch to the new image. Then I cleared all the browser cache and other data. The problem that occurred the second time became that after clicking Generate, the previous image The history of the picture disappears directly`
I can't replicate this issue at all. I even tried swapping language to Chinese just in case that was relevant (being the only obvious difference in your screenshot)
![image](https://github.com/user-attachments/assets/243f44ac-c542-47d6-b350-ebd7edb9bc3a)
It swaps to new image automatically, and fills up the recent images batch area automatically

So it'd be hard to say what's going wrong.

Check if your browser console has errors? (hit F12)

**Comment by 1025307336** [comment]
@mcmonkey4eva 

This is the console error


[Deprecation]The 'textprediction' attribute will be removed in the future. Use the 'writingsuggestions' attribute instead. See https://learn.microsoft.com/en-us/microsoft-edge/web-platform/site-impacting- changes for more information.

After I cleared all the data in IE and edge browsers, it still didn't work. However, when I opened the Google browser, the automatic image switching button was switched normally, and the default language was English interface.

Then I used the edge browser and everything returned to normal.

Hopefully there will be an option in the future to not use last browser data.

This function seems to be very easy to cause bugs. In the past, there was a problem that nothing was displayed in the image history tab below, not even the path was displayed.
